### PR TITLE
Fix #397 (wrong date handling)

### LINKED ISSUE #397: Date handling inconsistent
when getting the filename the date field of the entry is casted from a moment instance to a Date instance.

See https://github.com/cozy-labs/konnectors/blob/master/server/lib/naming.coffee#L31

Just getting the filename should not modify the entry.

:warning:  Each method using entry.date after that must expect a moment instance in https://github.com/cozy-labs/konnectors/blob/master/server/lib/save_data_and_file.coffee

### PR DESCRIPTION
**Pull request by nicofrand**

### CODE REVIEW

File: server/lib/naming.coffee [hunk 1]
```diff
@@ -26,20 +25,17 @@ module.exports =
     #
     getEntryFileName: (entry, options) ->
         name = ""
-        # We reset the date modified in save_data_and_file
-        if entry.date?
-            entry.date = new Date(entry.date)
-        if typeof(options) is "string"
-            name = "#{entry.date.format 'YYYYMM'}_#{options}.pdf"
+        date = entry.date
```
Comment (ZeHiro) [inline server/lib/naming.coffee]: can you maintain the test on existence of entry.date?

File: server/lib/save_data_and_file.coffee [hunk 2]
```diff
@@ -20,6 +20,11 @@ module.exports = (log, model, options, tags) ->
 
         # For each entry...
         async.eachSeries entriesToSave, (entry, callback) ->
+            if (entry.date not instanceof moment)
+                log.error 'Moment instance expected for date field'
+                log.info 'Bill creation aborted'
+                return callback()
```
Comment (ZeHiro) [inline server/lib/save_data_and_file.coffee]: can you pass an error message to the callback?

Comment (nicofrand) [inline server/lib/save_data_and_file.coffee]: What if I remove the log.error and then return the same message in the callback ?

Comment (ZeHiro) [inline server/lib/save_data_and_file.coffee]: Seems to be a good compromise. Watching more carefully the rest of the code, the errors are never shown to the user, you can keep it like that.

**Comment by nicofrand** [comment]
I'll rebase it

**Comment by nicofrand** [comment]
LGTM. @ZeHiro any comment ?

**Comment by ZeHiro** [comment]
@nicofrand nope

**Pull request merged**